tycho-apitools-plugin/src/main/java/org/eclipse/tycho/apitools/ApiAnalysisMojo.java

```diff
@@ -141,6 +145,30 @@ public class ApiAnalysisMojo extends AbstractMojo {
 	@Parameter(defaultValue = "false", property = "tycho.apitools.enhanceLogs")
 	private boolean enhanceLogs;
 
+	/**
+	 * Whether to create an XML report for the problems like the one created by the
+	 * <a href=
+	 * "https://help.eclipse.org/latest/topic/org.eclipse.pde.doc.user/reference/api-tooling/ant-tasks/analysis-ant-task.htm">API
+	 * Tools ANT task</a>.
+	 * <p>
+	 * Typically, this is best used in combination with
+	 * {@code <failOnError>false</failOnError>} in order to not fail the Maven build
+	 * and allow subsequent analysis by external means.
+	 *
+	 * @since 4.0.13
+	 */
+	@Parameter(defaultValue = "false", property = "tycho.apitools.writeXmlReport")
+	private boolean writeXmlReport;
```

Comment:
I don't think we need an extra boolean here, the usual default with maven is that one writes a report and then it might not get used, given that I assume its quite fast to generate the xml.

We might skip writing the report if the user sets the result file to an explicit empty string.

Reply:
I'm fine with either way. The only important thing to consider is that if `<failOnError>true</failOnError>` is enabled and the build fails immediately after any bundle has a problem the XML files are not all that useful, since you won't get results from all the other bundles. Hence also the javadoc comment.

Reply:
Can we maybe just write the report before the check for errors (and fail the mojo) then one should get a report for the current module (what is usual maven behavior with failing tests as well)